app/models/leave_application.rb

```diff
@@ -39,7 +39,8 @@ class LeaveApplication < ApplicationRecord
     end
 
     event :cancel, after: :return_leave_time_usable_hours do
-      transitions to: :canceled, from: [:pending, :approved, :rejected]
+      transitions to: :canceled, from: [:pending, :rejected]
+      transitions to: :canceled, from: :approved, unless: :happened?
```

Comment:
Add `Guard` to change `status` from `:approved` to `:canceled` only if `leave_application` not yet hapenning.